File: concrete/src/main/java/pcb/uwu/unit/base/StoneUnit.java
package pcb.uwu.unit.base;

import java.math.BigDecimal;

public class StoneUnit extends MassUnit {

	private static final BigDecimal KILOGRAMS_PER_STONE = new BigDecimal("6.35029318");

	public static final StoneUnit STONE = new StoneUnit();

	private StoneUnit() {
		super("st", "stone", "stones",
				value -> value.times(KILOGRAMS_PER_STONE),
				value -> value.div(KILOGRAMS_PER_STONE));
	}
}

File: implementation/src/main/java/pcb/uwu/unit/base/StoneUnit.kt
package pcb.uwu.unit.base

import pcb.uwu.unit.base.StoneUnit.CONVERT

object StoneUnit
    : MassUnit(symbol = "st",
               singularName = "stone",
               pluralName = "stones",
               translationToCanonical = { it * CONVERT },
               translationFromCanonical = { it / CONVERT })
{
    const val CONVERT = 6.35029318
}
